src/Symfony/Component/Yaml/Inline.php

```diff
@@ -659,9 +661,22 @@ private static function evaluateScalar($scalar, $flags, $references = array())
 
                         return $time;
                 }
-            default:
-                return (string) $scalar;
         }
+
+        if (null !== self::$tagResolver && '!' === $scalar[0]) {
+            $tagLength = strcspn($scalar, " \t", 1);
+            $tag = substr($scalar, 1, $tagLength);
+            $i = 0;
+            $scalar = self::parseScalar(ltrim(substr($scalar, $tagLength)), $flags, null, array('"', "'"), $i, false, $references);
+
+            try {
+                return self::$tagResolver->resolve($scalar, $tag);
+            } catch (UnsupportedTagException $e) {
+                // Ignored for bc
```

Comment:
As I removed the exception in the last commit (and its catch), I think we can do it later to focus on the feature.